### PR TITLE
feat(editor): Overhaul expression editor modal

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

File: packages/editor-ui/src/components/NodeDetailsView.vue [hunk 1]
```diff
@@ -254,7 +254,7 @@ export default mixins(
 			return !this.readOnly && this.isTriggerNode && (isWebhookBasedNode || isPollingNode || override);
 		},
 		workflow(): Workflow {
-			return this.getCurrentWorkflow();
+			return this.getCurrentWorkflow(true);
```
Comment (ivov) [inline packages/editor-ui/src/components/NodeDetailsView.vue:257]: https://www.notion.so/n8n/P1-Reviews-Expression-editor-e049831cb7b34f3d8c54bc6cddb522e8#b8a823877a474eb2952bbcb2db11d905

File: packages/editor-ui/src/components/ParameterInputHint.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <template>
 	<n8n-text size="small" color="text-base" tag="div" v-if="hint">
-		<div v-if="!renderHTML" :class="{[$style.hint]: true, [$style.highlight]: highlight}">{{ hint }}</div>
-		<div v-else ref="hint" :class="{[$style.hint]: true, [$style.highlight]: highlight}" v-html="sanitizeHtml(hint)"></div>
+		<div v-if="!renderHTML" :class="{[$style.hint]: true}">{{ hint }}</div>
+		<div v-else ref="hint" :class="{[$style.hint]: true}" v-html="sanitizeHtml(hint)"></div>
```
Comment (ivov) [inline packages/editor-ui/src/components/ParameterInputHint.vue:4]: https://www.notion.so/n8n/P1-Reviews-Expression-editor-e049831cb7b34f3d8c54bc6cddb522e8#c8171ae8a3784327bff8bf368f486faf

File: packages/editor-ui/src/components/VariableSelector.vue [hunk 3]
```diff
@@ -537,13 +537,15 @@ export default mixins(
 				if (parentNode.length) {
 					// If the node has an input node add the input data
 
+					const activeInputParentNode = parentNode.find(node => node === this.ndvStore.ndvInputNodeName)!;
```
Comment (ivov) [inline packages/editor-ui/src/components/VariableSelector.vue:540]: https://www.notion.so/n8n/P1-Reviews-Expression-editor-e049831cb7b34f3d8c54bc6cddb522e8#4bb85357c9294a6e88fc6a631669d786

File: packages/editor-ui/src/components/ExpressionEditorModal/n8nLanguagePack/index.cjs [hunk 4]
```diff
@@ -0,0 +1,57 @@
+'use strict';
```
Comment (ivov) [inline packages/editor-ui/src/components/ExpressionEditorModal/n8nLanguagePack/index.cjs:1]: Everything under `/n8nLanguagePack` is auto-generated externally. Will decide whether to include or publish in next phase.

File: packages/workflow/src/Expression.ts [hunk 5]
```diff
@@ -277,7 +277,16 @@ export class Expression {
 					throw error;
 				}
 			}
+
+			if (
+				error instanceof Error &&
+				typeof error.message === 'string' &&
+				error.name === 'SyntaxError'
+			) {
+				throw new Error(error.message);
```
Comment (mutdmour) [inline packages/workflow/src/Expression.ts:286]: Why is this check needed if it's throwing same error with same message?

File: packages/editor-ui/src/components/ParameterInputHint.vue (hunk 2 above)
Comment (mutdmour) [reply packages/editor-ui/src/components/ParameterInputHint.vue:4]: I am confused.. are you sure this is what Max wanted here? to remove highlight completely? I think he wanted to fix issue when switching.. 
also if you are going to remove this here, you need to complete the clean up.. change class back to `class="hint"` and remove the prop and where it gets passed

File: packages/editor-ui/src/components/NodeDetailsView.vue (hunk 1 above)
Comment (mutdmour) [reply packages/editor-ui/src/components/NodeDetailsView.vue:257]: this is an expensive operation to create new workflow.. and was huge performance benefit to get rid of it everytime we open NDV or make an operation.. is there a better fix here possible?

File: packages/editor-ui/src/components/ParameterInputWrapper.vue [hunk 6]
```diff
@@ -157,23 +157,14 @@ export default mixins(
...
-					if (!inputData || (inputData && inputData.length <= 1)) {
-						return this.expressionValueComputed;
-					}
-
-					return this.$locale.baseText(`parameterInput.expressionResult`, {
-						interpolate: {
-							result: this.expressionValueComputed,
-						},
-					});
+					return this.expressionValueComputed;
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/components/ParameterInputWrapper.vue:167]: are you sure this is what Max wanted to lose this behavior completely?

**Review by mutdmour — COMMENTED** [review]
not done reviewing but please take a look

File: packages/editor-ui/src/components/ParameterInputHint.vue (hunk 2 above)
Comment (ivov) [reply packages/editor-ui/src/components/ParameterInputHint.vue:4]: When he flagged, he identified these as two issues color switch and `e.g.`. I commented on both and he confirmed, see thread. Else we confirm again?
image

File: packages/editor-ui/src/components/ParameterInputWrapper.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (ivov) [reply packages/editor-ui/src/components/ParameterInputWrapper.vue:167]: Same as https://github.com/n8n-io/n8n/pull/4631#discussion_r1035837118

File: packages/workflow/src/Expression.ts (hunk 5 above)
Comment (ivov) [reply packages/workflow/src/Expression.ts:286]: To surface the `SyntaxError`, preventing it from becoming `null`.

File: packages/editor-ui/src/components/ParameterInputHint.vue (hunk 2 above)
Comment (mutdmour) [reply packages/editor-ui/src/components/ParameterInputHint.vue:4]: reading through the thread but does not seem to be saying remove it.. please double check

File: packages/editor-ui/src/components/NodeDetailsView.vue (hunk 1 above)
Comment (ivov) [reply packages/editor-ui/src/components/NodeDetailsView.vue]: Thanks. Moved the call to `saveAsNewWorkflow`.

File: packages/editor-ui/src/components/ParameterInputHint.vue (hunk 2 above)
Comment (ivov) [reply packages/editor-ui/src/components/ParameterInputHint.vue:4]: /cc @maxtkacz

File: packages/editor-ui/src/components/ParameterInputWrapper.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (ivov) [reply packages/editor-ui/src/components/ParameterInputWrapper.vue:167]: Confirmed.

File: packages/editor-ui/src/components/ParameterInputHint.vue (hunk 2 above)
Comment (ivov) [reply packages/editor-ui/src/components/ParameterInputHint.vue]: Reverted, see [66570b6](https://github.com/n8n-io/n8n/pull/4631/commits/66570b687581c098915f743666851b1b6ca0e099)

File: packages/editor-ui/src/components/ExpressionEditorModal/ExpressionModalOutput.vue [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+					segment.to = cursor;
+
+					return segment;
+				})
+				.filter((segment): segment is Resolved => segment.kind === 'resolvable');
+		},
+	},
+	methods: {
+		getValue() {
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/editor-ui/src/components/ExpressionEditorModal/ExpressionModalOutput.vue:87]: On line 211 of `ExpressionEdit`. Kept it as is.

File: packages/editor-ui/src/components/VariableSelectorItem.vue [hunk 8]
```diff
@@ -53,6 +53,20 @@ export default mixins(externalHooks).extend({
...
 	],
+	mounted() {
+		if (this.extended) return;
+
+		const shouldAutoExtend = [
+			this.$locale.baseText('variableSelectorItem.currentNode'),
+			this.$locale.baseText('variableSelectorItem.inputData'),
+			this.$locale.baseText('variableSelectorItem.binary'),
+			this.$locale.baseText('variableSelectorItem.json'),
+		].includes(this.item.name) && this.item.key === undefined;
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/editor-ui/src/components/VariableSelectorItem.vue:64]: Agree, added as tech debt: https://linear.app/n8n/issue/N8N-5799/find-a-way-to-id-variable-selector-items-without-using-name

File: packages/editor-ui/src/components/ExpressionEditorModal/ExpressionModalInput.vue [hunk 9]
```diff
@@ -33,6 +33,7 @@ export default mixins(workflowHelpers).extend({
 		return {
 			editor: null as EditorView | null,
 			errorsInSuccession: 0,
+			isReadOnly: window !== window.parent && window.parent.location.pathname.includes('/executions'),
```
Comment (mutdmour) [inline packages/editor-ui/src/components/ExpressionEditorModal/ExpressionModalInput.vue:36]: not a great solution.. cannot we simply pass it from parameter input? we already have that info there no?

**Review by mutdmour — COMMENTED** [review]
Looks good. just one last question

**Review by mutdmour — DISMISSED** [review]

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.206.1`

**Pull request merged**